### PR TITLE
feat: added s3 file download support

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**

### CODE REVIEW

File: s3downloader.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import logging
```
Comment (asafMasa) [inline s3downloader.py]: try use our logger

File: s3downloader.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+log = logging.getLogger('s3_downloader')
+log.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+# create formatter and add it to the handlers
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+ch.setFormatter(formatter)
+# add the handlers to the logger
+log.addHandler(ch)
+
+endpoint_url = os.environ.get('S3_ENDPOINT_URL', 'http://localhost:9000')
```
Comment (asafMasa) [inline s3downloader.py:5]: add readme to explain how it works and what keys need to be supplied

File: s3downloader.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+aws_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID', '')
+aws_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY', '')
+bucket = os.environ.get('S3_BUCKET', '')
+object_key = os.environ.get('S3_OBJECT_KEY', 'mapproxy.yaml')
```
Comment (asafMasa) [inline s3downloader.py:10]: check if ssl_enabled is needed with @AnguIar

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: s3downloader.py (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply s3downloader.py]: fixed in b45b9bb

File: s3downloader.py (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply s3downloader.py:5]: fixed in b45b9bb

File: s3downloader.py (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply s3downloader.py:10]: fixed in 8090fbb

File: s3downloader.py [hunk 4]
```diff
@@ -1,36 +1,50 @@
...
-log.setLevel(logging.DEBUG)
-ch = logging.StreamHandler()
-# create formatter and add it to the handlers
-formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
-ch.setFormatter(formatter)
-# add the handlers to the logger
-log.addHandler(ch)
+
+
+log = generate_logger('s3-file-downloader', log_level='DEBUG',
```
Comment (asafMasa) [inline s3downloader.py]: change to environment variable

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.6-alpine3.12
+RUN addgroup -S app && adduser -S app -G app
+RUN mkdir /app
+WORKDIR /app
+RUN mkdir downloads
+RUN apk update -q --no-cache \
+    && apk add -q --no-cache python3 py3-pip
+COPY requirements.txt ./
+RUN pip3 install -r ./requirements.txt -t /app
```
Comment (asafMasa) [inline Dockerfile:8]: make sure it works in openshift

File: s3downloader.py [hunk 6]
```diff
@@ -1,36 +1,50 @@
...
+
+
 """
 This method download object into local file system.
 """
+
+
 try:
+    ssl_enabled = to_bool(use_ssl, False)
+    print(ssl_enabled)
```
Comment (asafMasa) [inline s3downloader.py]: remove print

File: s3downloader.py [hunk 7]
```diff
@@ -1,36 +1,50 @@
...
                                 aws_access_key_id=aws_access_key_id,
-                                aws_secret_access_key=aws_secret_access_key)
+                                aws_secret_access_key=aws_secret_access_key,
+                                use_ssl=ssl_enabled)
 
     resource.Bucket(bucket).download_file(object_key, destination)
     log.debug(f'Successfully downloaded file to: {destination}')
 except Exception as e:
-    log.error(f'Error occurred while trying download file: {e}')
+    log.debug(f'Error occurred while trying download file: {e}')
```
Comment (asafMasa) [inline s3downloader.py]: why it is not error?

**Review by asafMasa — CHANGES_REQUESTED** [review]
:+1:

File: s3downloader.py (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply s3downloader.py]: fixed in cd5abe4

File: Dockerfile (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply Dockerfile:8]: fixed in 9c69a55

File: s3downloader.py (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply s3downloader.py]: fixed in 5d5846d

File: s3downloader.py (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply s3downloader.py]: fixed in 93ec761

**Review by asafMasa — APPROVED** [review]

**Pull request merged**